### DISCUSSION TITLE
vsketch is not launching for me

### DISCUSSION THREAD

**Discussion by nicholaserik** [abey79/vsketch#105] [Q&A]
Hi Antoine,

Thank you for all the work you have put into this. 
I'm running MacOS 11.2.3 (Big Sur) and have been trying to use vsketch but have not been able to launch it.
When I run your example file or the example you have in your readme the python launcher comes up in my dock but does not open anything. Activity monitor says that it is running at 100% CPU and Finder says that python is not responding. I have tried uninstalling deleting and reinstalling vsketch and venv with the same results. Do you have any ideas of what else I should be trying? 

update: I was able to get it running on my mac by running pip3 install . 
But now when I copy your code for Schotter it opens the vpype viewer window and says Error (see console). The console only says: "qt.qpa.window: <QNSWindow: 0x7f8fc6c06b50; contentView=<QNSView: 0x7f8fc8107160; QCocoaWindow(0x7f8fc8107000, window=QWidgetWindow(0x7f8fc8106b00, name="QWidgetClassWindow"))>> has active key-value observers (KVO)! These will stop working now that the window is recreated, and will result in exceptions when the observers are removed. Break in QCocoaWindow::recreateWindowIfNeeded to debug." It repeats this three times. 
When I run your examples/schotter it runs correctly. The difference I see between the two is self. or vsk. I thought I had read somewhere that you had changed vsk. to self. but maybe I'm remembering that incorrectly.

I just saw your post about changing to vsk and sketchclass. Thank you.

Best,
nicholas.

**Comment by abey79** [comment]
Hi @nicholaserik,

I've heard about several issues with Big Sur, but haven't been able to properly reproduce them. My advice would be to use Python 3.9.2 from [MacPort](https://www.macports.org) and install vsketch in a venv which you must make 100% has been created by that Python install in particular (the best way is to use you the full path:  `/opt/local/bin/python3.9 -m venv venv`).

I'm happy to assist with the details if needed. Dont hesitate to PM me on discord if needed.

Best,
Antoine

**Comment by felixmeissner** [comment]
Hi @abey79, very excited about getting started with vsketch but struggeling to get it installed. Running in this error: 

ERROR: Package 'vsketch' requires a different Python: 3.11.0 not in '<3.11,>=3.8'

Any idea how to overcome this?  Thanks for your help in advance.
Felix

**Reply by abey79** [reply]
Hi @felixmeissner

*vpype* and *vsketch* don't currently support Python 3.11. This will be fixed by the next release of *vpype*. In the mean time, you can use Python 3.10. It's ok to install python 3.10 alongside your current installation of python 3.11. You will need to be mindful of which Python version is used upon installation and I advise to use the full path to avoid any mistake.

For pipx-based installation:
```bash
 pipx install "git+https://github.com/abey79/vsketch" --python /full/path/to/python3.10
```
The actual full path depends on how you installed Python. I can help if tell me more about your platform and method of Python installation.

**Reply by felixmeissner** [reply]
thanks for super speedy reply. 

I used visual studio code and downloaded python through extension marketplace but also downloaded selected versions from python.org. I tried multiple different universal versions but always (from 3.11.X to 3.7.X) but always end up after `pip install . `in the same error: `ERROR: Package 'vsketch' requires a different Python: 3.11.0 not in '<3.11,>=3.8' - I am new to all this but followed this tutorial [here](https://www.youtube.com/watch?v=9NQVRFnkb1E). So at this stage I thought I ping here.

**Reply by abey79** [reply]
I wasn't aware of this video! Unfortunately, it uses the "old" way to install vsketch. I strongly advise to use pipx instead. The [vpype installation instruction](https://vpype.readthedocs.io/en/latest/install.html) provide a bit more details which are all relevant to installing vsketch.

In a nutshell, you first need to install pipx:

```
/path/to/python3.10 -m pip install --user pipx
/path/to/python3.10 -m pipx ensurepath
```

This should make the `pipx` command available (you may have to restart your shell for that). Then you can install vsketch with the command in my previous message.

**Comment by felixmeissner** [comment]
thanks. I now set up pipx but still now success :-(

[image: Screen Shot 2022-11-06 at 23.00.09.png]

same error. do you think it has to do with zsh?

Am So., 6. Nov. 2022 um 22:38 Uhr schrieb Antoine Beyeler <
***@***.***>:

> I wasn't aware of this video! Unfortunately, it uses the "old" way to
> install vsketch. I strongly advise to use pipx instead. The vpype
> installation instruction
> <https://vpype.readthedocs.io/en/latest/install.html> provide a bit more
> details which are all relevant to installing vsketch.
>
> In a nutshell, you first need to install pipx:
>
> /path/to/python3.10 -m pip install --user pipx
> /path/to/python3.10 -m pipx ensurepath
>
> This should make the pipx command available (you may have to restart your
> shell for that). Then you can install vsketch with the command in my
> previous message.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/abey79/vsketch/discussions/105#discussioncomment-4070859>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA64AQ55PU6XHFNEBK75WODWHAQMPANCNFSM4ZTROFIA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
--
Follow me: @felix.meissner <https://instagram.com/felix.meissner>
Mobil: +31 615 22 75 60
E-mail: ***@***.***

**Reply by abey79** [reply]
Your message is unfortunately badly formatted, so I can't answer.
